src/queue.c

```diff
@@ -773,7 +773,7 @@ _dispatch_async_redirect_invoke(dispatch_continuation_t dc,
 {
 	dispatch_thread_frame_s dtf;
 	struct dispatch_continuation_s *other_dc = dc->dc_other;
-	dispatch_invoke_flags_t ctxt_flags = (dispatch_invoke_flags_t)dc->dc_ctxt;
+	dispatch_invoke_flags_t ctxt_flags = (dispatch_invoke_flags_t)(uintptr_t)dc->dc_ctxt;
```

Comment:
This change resolves this warning:

```
S:\SourceCache\swift-corelibs-libdispatch\src\queue.c(776,39): warning: cast to smaller integer type 'dispatch_invoke_flags_t' from 'void *' [-Wvoid-pointer-to-enum-cast]
        dispatch_invoke_flags_t ctxt_flags = (dispatch_invoke_flags_t)dc->dc_ctxt;
                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

The compiler doesn't know that `dc_ctxt` here was originally a value of type `dispatch_invoke_flags_t`, so it thinks it could be any pointer. `dispatch_invoke_flags_t` is a `uint32_t`. Casting from a pointer to a narrower integer type is UB. Casting to `uintptr_t` is always defined, then casting from `uintptr_t` to `uint32_t` is truncating.